lib/offsite_payments/integrations/universal.rb

```diff
@@ -78,7 +78,8 @@ def sign_fields
         end
 
         def generate_signature
-          Universal.sign(@fields, @key)
+          fields_to_sign = @fields.select { |key, _| key.start_with?('x_') && key != 'x_signature' }
```

Comment:
Are you 100% sure all keys are strings, and can't be symbols? I'm not familiar with this library enough, but I see things like `add_fields` which do not stringify all inputs.